aten/src/ATen/native/mps/operations/UnaryOps.mm

```diff
@@ -176,5 +178,160 @@ void unary_op(const Tensor& self, const Tensor& output, std::string op_name, Una
+                                                              shape: newCachedGraph->inputTensor_.shape
+                                                              dataType:realPartTensor.dataType];                
+
+                MPSGraphTensor* isRealZero = [mpsGraph equalWithPrimaryTensor:realPartTensor
+                                                              secondaryTensor:zeroTensor
+                                                              name: nil];
+
+                MPSGraphTensor* isImaginaryZero = [mpsGraph equalWithPrimaryTensor:imaginaryPartTensor
+                                                              secondaryTensor:zeroTensor
+                                                              name: nil];
+
+                MPSGraphTensor* isComplexZero = [mpsGraph logicalANDWithPrimaryTensor:isRealZero
+                                                              secondaryTensor:isImaginaryZero
+                                                              name: nil];
+
+                MPSGraphTensor* sgnDenomReal = [mpsGraph squareWithTensor:realPartTensor
+                                                              name: nil];
+
+                MPSGraphTensor* sgnDenomImaginary = [mpsGraph squareWithTensor:imaginaryPartTensor
+                                                              name: nil];
+
+                MPSGraphTensor* sgnDenomSum = [mpsGraph additionWithPrimaryTensor:sgnDenomReal
+                                                              secondaryTensor:sgnDenomImaginary
+                                                              name: nil];
+
+                MPSGraphTensor* sgnDenom = [mpsGraph squareRootWithTensor:sgnDenomSum
+                                                              name: nil];
+
+                MPSGraphTensor* sgnRealTensor = [mpsGraph divisionWithPrimaryTensor:realPartTensor
+                                                              secondaryTensor:sgnDenom
+                                                              name: nil];
+
+                MPSGraphTensor* sgnImaginaryTensor = [mpsGraph divisionWithPrimaryTensor:imaginaryPartTensor
+                                                              secondaryTensor:sgnDenom
+                                                              name: nil];
+
+                MPSGraphTensor* sgnComplexTensor = [mpsGraph concatTensors:@[sgnRealTensor, sgnImaginaryTensor]
+                                                              dimension: 1
+                                                              name: nil];
+
+                sgnTensor = [mpsGraph selectWithPredicateTensor:isComplexZero
+                                                              truePredicateTensor:complexZeroTensor
+                                                              falsePredicateTensor:sgnComplexTensor
+                                                              name:nil];
+              } else {
```

Comment:
if a non-complex number is used then return matrix with -1 for negative input, 1 for positive input, and 0 otherwise.